mmv1/templates/terraform/post_delete/private_cloud.go.erb

```diff
@@ -26,9 +26,19 @@ privateCloudPollRead := func(d *schema.ResourceData, meta interface{}) transport
                 RawURL: url,
                 UserAgent: userAgent,
             })
+            // if resource does not exist
             if err != nil {
+                log.Printf("[DEBUG] The Private cloud has been successfully deleted.")
                 return res, err
             }
+            // if resource exists but is marked for deletion
+            log.Printf("[DEBUG] Fetching state of the private cloud.")
+            v, ok := res["state"]
+            if ok && v.(string) == "DELETED" {
+                log.Printf("[DEBUG] The Private cloud has been successfully marked for delayed deletion.")
+                return res, errors.New("PrivateCloud is marked for deletion")
```

Comment:
I think you might want to return nil, nil here to signify that the resource was successfully put in a deleted state.